### PR TITLE
Changes by Assaf

### PR DESCRIPTION
**Pull request by assafar26**
hey
i added my endpoints, app is running on this branch, i need @niv-e  help with the mapper because i added a department field the users entity and boundary, in addition i will implement the updateUserDepartment tomorrow and the getUserByMinimumage function also. 
@LiorAriely don't stress i made a few upgrades to your code, also merged the getbydomain to my method. 
i had to change the getuserbycriteria path against eyal's SPEC because it overrides the getAllUsers path and make it disappear in the swagger. 
good luck

### CODE REVIEW

File: ReactiveUsersMicroservice/src/main/java/Cloud2024/ReactiveUsersMicroservice/Infrastructure/UserMapper.java [hunk 1]
```diff
@@ -22,6 +22,7 @@ public ModelMapper modelMapper(){
                     mapper.map(src -> generateRandomAsterisks(), UserBoundary::setPassword);
                     mapper.map(UserEntity::getFirstName, (dest, v) -> dest.getName().setFirst((String) v));
                     mapper.map(UserEntity::getLastName, (dest, v) -> dest.getName().setLast((String) v));
+                    mapper.map(UserEntity::getDepartment, UserBoundary::setDepartment); // Map department
```
Comment (niv-e) [inline ReactiveUsersMicroservice/src/main/java/Cloud2024/ReactiveUsersMicroservice/Infrastructure/UserMapper.java]: The department property should be map autos like all the other properties.
Here you see only properties that need customization to their mapping.

File: ReactiveUsersMicroservice/src/main/java/Cloud2024/ReactiveUsersMicroservice/Infrastructure/UserMapper.java [hunk 2]
```diff
@@ -30,6 +31,7 @@ public ModelMapper modelMapper(){
                 mapper -> {
                     mapper.map(src -> src.getName().getFirst(), UserEntity::setFirstName);
                     mapper.map(src -> src.getName().getLast(), UserEntity::setLastName);
+                    mapper.map(UserBoundary::getDepartment, UserEntity::setDepartment); // Map department
```
Comment (niv-e) [inline ReactiveUsersMicroservice/src/main/java/Cloud2024/ReactiveUsersMicroservice/Infrastructure/UserMapper.java]: Same like the above

File: ReactiveUsersMicroservice/src/main/java/Cloud2024/ReactiveUsersMicroservice/Presentation/UserController.java [hunk 3]
```diff
@@ -29,6 +28,35 @@ public Mono<UserBoundary> createUser (
                 .log();
     }
 
+    @GetMapping(
+            path = "/criteria",
+            produces = {MediaType.TEXT_EVENT_STREAM_VALUE})
+    public Flux<UserBoundary> getUsersByCriteria(
```
Comment (niv-e) [inline ReactiveUsersMicroservice/src/main/java/Cloud2024/ReactiveUsersMicroservice/Presentation/UserController.java:34]: It's looks good, I am not sure the the GetAllUsers that what the led to the url conflicts.
I suggest you the comment all the end point and enable them one by one to find which one is making the problem

**Review by niv-e — APPROVED** [review]

File: ReactiveUsersMicroservice/src/main/java/Cloud2024/ReactiveUsersMicroservice/Infrastructure/UserMapper.java [hunk 4]
```diff
@@ -22,6 +22,7 @@ public ModelMapper modelMapper(){
                     mapper.map(src -> generateRandomAsterisks(), UserBoundary::setPassword);
                     mapper.map(UserEntity::getFirstName, (dest, v) -> dest.getName().setFirst((String) v));
                     mapper.map(UserEntity::getLastName, (dest, v) -> dest.getName().setLast((String) v));
+//                    mapper.map(src -> src.getDeptId(), UserBoundary::setDepartment);
```
Comment (assafar26) [inline ReactiveUsersMicroservice/src/main/java/Cloud2024/ReactiveUsersMicroservice/Infrastructure/UserMapper.java:25]: @niv-e having trouble with the mapper, in UserBoundary there is a Department field that is supposed to be converted to deptId in the UserEntity. i am having trouble implpementing it in the mapper.

**Pull request merged**